File: circle.cpp
#include "circle.h"

// constructor
Circle::Circle() {
    center_x_ = 0;
    center_y_ = 0;
    radius_ = 0;
}

// destructor
Circle::~Circle() {
    center_x_ = 0;
    center_y_ = 0;
    radius_ = 0;
}

// input data for circle
void Circle::Input(ifstream &ifst) {
    ifst >> center_x_ >> center_y_ >> radius_;
}

// get random data for circle
void Circle::InputRandom() {
    center_x_ = Random();
    center_y_ = Random();
    radius_ = Random();
}

// output circle data
void Circle::Output(ofstream &ofst) {
    ofst << " Circle: center_x = " 
         << center_x_ << ", center_y = " << center_y_
         << ", radius = " << radius_
         << ". Perimeter = " << Perimeter() << "\n";
}

// get perimeter of circle
double Circle::Perimeter() {
    double pi = 3.14159265359;
    return 2 * pi * radius_;
}

File: circle.py
from shape import Shape
import math


class Circle(Shape):
    def __init__(self, color, line):
        Shape.__init__(self, color)
        self.radius = 0
        self.is_valid = True
        self.input(line)

    def input(self, data):
        Shape.set(self, data, 1)
        self.radius = data[2]
        if self.radius < 0:
            self.is_valid = False

    def output(self, file):
        out_str = (self.color + " Circle: " +
                                " center_x = " + str(self.nodes[0].X) +
                                " center_y = " + str(self.nodes[0].Y) +
                                " radius = " + str(self.radius) +
                                " Perimeter = " + str(self.get_perimeter()) + "\n")
        file.write(out_str)
        if not self.is_valid:
            out_str = "This Circle is invalid\n"
            file.write(out_str)

    def get_perimeter(self):
        return 2 * math.pi * self.radius

